tests/test_fixtures.py

```diff
@@ -33,6 +33,19 @@ def test_json_response(self, client, accept_json):
         res = client.get(url_for('ping'), headers=accept_json)
         assert res.json == {'ping': 'pong'}
 
+    def test_json_response_compare_to_status_code(self, client, accept_json):
+        assert client.get(url_for('ping'), headers=accept_json) == 200
+        assert client.get('fake-route', headers=accept_json) == 404
+        res = client.get(url_for('ping'), headers=accept_json)
+        assert res == res
+
+    @pytest.mark.xfail(
```

Comment:
xfail should be used to mark tests as expected failures or flaky. I believe the proper way here is to use `pytest.raises` and probably check the exception message:

```python
def test_mismatching_eq_comparison(self, client, accept_json):
    with pytest.raises(AssertionError, match='Mismatch in status code for response: 404 != 200'):
        assert client.get('fake-route', headers=accept_json) == 200
```